.github/workflows/build_and_functional_tests.yml

```diff
@@ -0,0 +1,78 @@
+name: Build and run functional tests using ragger through reusable workflow
+
+# This workflow will Lint, Build and then Run tests.
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+      - develop
+    tags:
+      - '*'
+  pull_request:
+
+jobs:
+  Build:
+    strategy:
+      matrix:
+        os: [macos-latest, ubuntu-latest]
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - name: Install dependencies
+        run: sudo apt update && sudo apt install -qy eslint
+        if: runner.os == 'Linux'
+
+      - name: Clone
+        uses: actions/checkout@v4
+
+      - name: Install Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: node
+
+      - name: Packages installation
+        run: npm install
+
+      - name: Compile
+        run: npm run compile
+
+      - name: Linter
+        if: runner.os == 'Linux'
+        run: npm run lint
+
+      - name: Test with Linux
+        if: runner.os == 'Linux'
+        run: xvfb-run -a npm test
+
+      - name: Test
+        if: runner.os != 'Linux'
+        run: npm test
+
+  Build_Windows:
+    strategy:
+      matrix:
+        os: [windows-latest]
+    runs-on: ${{ matrix.os }}

```

Comment:
Does this job need a matrix if there's only one OS ?

Reply:
Of course not.
Fixed